modules/data-entry/src/main/frontend/src/questionnaire/FormPagination.jsx

```diff
@@ -178,56 +174,56 @@ function FormPagination (props) {
       (doneLabel || 'Save')}
     </Button>
 
-  let stepper = (isBack) =>
-    <MobileStepper
-      variant="progress"
-      // Offset back bar 1 to create a "current page" region.
-      // If the final page has been saved, progress the front bar to complete
-      activeStep={activePage + (isBack ? 1 : (lastSaveStatus && savedLastPage ? 1 : 0))}
-      // Change the color of the back bar
-      LinearProgressProps={isBack ? {classes: {barColorPrimary: classes.formStepperTopBar}}: null}
-      // Hide the backround of the front bar to segment of back bar
-      className={`${classes.formStepper} ${isBack && classes.formStepperBottom} ${!fullScreen && classes.formStepperFullScreen}`}
-      classes={isBack ? null : {progress:classes.formStepperBottomBackground}}
-      // base 0 to base 1, plus 1 for the "current page" region
-      steps={lastValidPage() + 2}
-      nextButton={saveButton}
-      backButton={
-        lastValidPage() > 0
-          ? <Button
-              type="submit"
-              // Don't disable until form submission started
-              disabled={(activePage === 0 && !pendingSubmission)
-                || saveInProgress
-                || lastSaveStatus === false}
-              onClick={handleBack}
-              className={classes.paginationButton}
-              variant="outlined"
-            >
-              Back
-            </Button>
-          : null
-      }
-    />
+  let backButton =
+    <Button
+      type="submit"
+      variant="outlined"
+      // Don't disable until form submission started
+      disabled={(activePage === 0 && !pendingSubmission)
+        || saveInProgress
+        || lastSaveStatus === false}
+      className={classes.paginationButton}
+      onClick={handleBack}
+    >
+      Back
+    </Button>
 
   return (
-    paginationEnabled ?
-    lastValidPage() > 0 ?
-      <React.Fragment>
-        {/* Back bar to show a different colored current page section*/}
-        {stepper(false)}
-        {/* Front bar to color completed pages differently from current page */}
-        {stepper(true)}
-      </React.Fragment>
-    :
-      saveButton
+    enabled
+    ?
+      lastValidPage() > 0
+      ?
+        <MobileStepper
+          variant="progress"
+          // Offset back bar 1 to create a "current page" region.
+          // If the final page has been saved, progress the front bar to complete
+          activeStep={activePage + (lastSaveStatus && savedLastPage ? 1 : 0)}
+          // Change the color of the back bar
+          LinearProgressProps={{ 
+              classes: { barColorPrimary: classes.formStepperTopBar,
+                         bar2Buffer: classes.completedProgressBar,
+                         dashed: classes.bufferProgressBar
+                       },
```

Comment:
Looks like I misinterpreted the meanings in my previous comment. Let's rename `completedProgressBar` to `formStepperBufferBar` and `bufferProgressBar` to `formStepperBackgroundBar`.

Also, `classes.formStepperTopBar` doesn't seem to be defined, so it can be removed from here.

```suggestion
              classes: {
                         bar2Buffer: classes.formStepperBufferBar,
                         dashed: classes.formStepperBackgroundBar
                       },
```